File: src/main/java/com/company/frontstudiodemo/entity/TestRegistration.java
package com.company.frontstudiodemo.entity;

import io.jmix.core.entity.annotation.JmixGeneratedValue;
import io.jmix.core.metamodel.annotation.InstanceName;
import io.jmix.core.metamodel.annotation.JmixEntity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.UUID;

@JmixEntity
@Table(name = "TEST_REGISTRATION")
@Entity
public class TestRegistration {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private UUID id;

    @InstanceName
    @Column(name = "NAME")
    private String name;

    @Column(name = "DESCRIPTION")
    private String description;

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }
}

File: frontend/src/jmix/entities/TestRegistration.ts
export class TestRegistration {
  static NAME = "TestRegistration";
  id?: string;
  name?: string | null;
  description?: string | null;
}
export type TestRegistrationViewName = "_base" | "_instance_name" | "_local";
export type TestRegistrationView<
  V extends TestRegistrationViewName
> = V extends "_base"
  ? Pick<TestRegistration, "id" | "name" | "description">
  : V extends "_instance_name"
  ? Pick<TestRegistration, "id" | "name">
  : V extends "_local"
  ? Pick<TestRegistration, "id" | "name" | "description">
  : never;
